paymentsheet/src/main/java/com/stripe/android/lpmfoundations/paymentmethod/definitions/CardDefinition.kt

```diff
@@ -31,39 +37,51 @@ internal object CardDefinition : PaymentMethodDefinition {
     override fun uiDefinitionFactory(): UiDefinitionFactory = CardUiDefinitionFactory
 }
 
-private object CardUiDefinitionFactory : UiDefinitionFactory.RequiresSharedDataSpec {
-    override fun createSupportedPaymentMethod(sharedDataSpec: SharedDataSpec) = SupportedPaymentMethod(
+private object CardUiDefinitionFactory : UiDefinitionFactory.Simple {
+    override fun createSupportedPaymentMethod() = SupportedPaymentMethod(
         paymentMethodDefinition = CardDefinition,
-        sharedDataSpec = sharedDataSpec,
         displayNameResource = R.string.stripe_paymentsheet_payment_method_card,
         iconResource = R.drawable.stripe_ic_paymentsheet_pm_card,
         tintIconOnSelection = true,
     )
 
     override fun createFormElements(
         metadata: PaymentMethodMetadata,
-        sharedDataSpec: SharedDataSpec,
-        transformSpecToElements: TransformSpecToElements
+        arguments: UiDefinitionFactory.Arguments,
     ): List<FormElement> {
         val billingDetailsCollectionConfiguration = metadata.billingDetailsCollectionConfiguration
-        val specs = listOf(
-            ContactInformationSpec(
-                collectName = false,
+        return buildList {
+            val contactInformationElement = contactInformationElement(
+                initialValues = arguments.initialValues,
                 collectEmail = billingDetailsCollectionConfiguration.collectsEmail,
                 collectPhone = billingDetailsCollectionConfiguration.collectsPhone,
-            ),
-            CardDetailsSectionSpec(
-                collectName = billingDetailsCollectionConfiguration.collectsName,
-            ),
-            CardBillingSpec(
-                collectionMode = billingDetailsCollectionConfiguration.address.toInternal(),
-            ),
-            SaveForFutureUseSpec(),
-        )
-        return transformSpecToElements.transform(
-            specs = specs,
-            replacePlaceholders = false,
-        )
+            )
+
+            if (contactInformationElement != null) {
+                add(contactInformationElement)
+            }
+
+            add(
+                CardDetailsSectionElement(
+                    context = arguments.context,
+                    initialValues = arguments.initialValues,
+                    identifier = IdentifierSpec.Generic("card_details"),
+                    collectName = billingDetailsCollectionConfiguration.collectsName,
+                    cbcEligibility = arguments.cbcEligibility,
+                )
+            )
+
+            val cardBillingElement = cardBillingElement(
+                billingDetailsCollectionConfiguration.address.toInternal(),
+                arguments.initialValues,
+                arguments.addressRepository,
+                arguments.shippingValues,
+            )
+            if (cardBillingElement != null) {
+                add(cardBillingElement)
```

Comment:
I think it'd be a bit easier to understand when the cardBillingElement and the contactInformationElement were included if the code was structured more like this:
```
if (collectionMode != BillingDetails...Never) {
   add(cardBillingElement(...))
}
```
what do you think of restructuring like that?